renku/service/views/datasets.py

```diff
@@ -343,3 +347,56 @@ def edit_dataset_view(user_data, cache):
             'warnings': warnings
         }
     )
+
+
+@use_kwargs(DatasetUnlinkRequest)
+@marshal_with(DatasetUnlinkResponseRPC)
+@header_doc('Unlink a file from a dataset', tags=(DATASET_BLUEPRINT_TAG, ))
+@dataset_blueprint.route(
+    '/datasets.unlink',
+    methods=['POST'],
+    provide_automatic_options=False,
+)
+@handle_base_except
+@handle_git_except
+@handle_renku_except
+@handle_validation_except
+@accepts_json
+@requires_cache
+@requires_identity
+def unlink_file_view(user_data, cache):
+    """Unlink a file from a dataset."""
+    ctx = DatasetUnlinkRequest().load(request.json)
+
+    include = ctx.get('include_filter')
+    exclude = ctx.get('exclude_filter')
+
+    user = cache.ensure_user(user_data)
+    project = cache.get_project(user, ctx['project_id'])
+
+    if ctx.get('commit_message') is None:
+        if include and exclude:
+            filters = '-I {0} -X {0}'.format(include, exclude)
+        elif not include and exclude:
+            filters = '-X {0}'.format(exclude)
+        else:
+            filters = '-I {0}'.format(include)
+
+        ctx['commit_message'] = (
+            'service: unlink dataset {0} {1}'.format(
+                ctx['short_name'], filters
+            )
+        )
+
+    with chdir(project.abs_path):
+        records = file_unlink(
+            short_name=ctx['short_name'],
+            include=ctx.get('include_filters'),
+            exclude=ctx.get('exclude_filters'),
+            yes=True,
+            interactive=False,
+            commit_message=ctx['commit_message']
+        )
+        unlinked = [record.name for record in records]
```

Comment:
Should we return `path` here instead of name to make the results uniquely identifiable?